core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -179,7 +180,11 @@ class HistoryServer(
   }
```

Comment:
I disagree that getApplicationList() can be removed because SparkUI class is still using this API though deleting it is just more pieces of code.

Yes, both original approach and view-based approach are suffering the risking of reading data from a mutable data source. However, I took a look at the code. The iterator (or the view) returned is from the mutable LinkedHashMap **applications** of FsHistoryProvider. It is only changed via FsHistoryProvider.mergeApplicationListing() by merging the original applications map with newly-updated apps to a new LinkedHashMap and change the applications reference to it. So the returned iterator or view is still binding to the old LinkedHashMap applications which is safe. I may be wrong, can you confirm?

I'll think about using the iterator returned from getApplicationList() as you said in the last comment. 

Thanks!